### PR TITLE
Changes from a 2 state rating to a 3 state rating (like netflix)

### PR DESCRIPTION
**Pull request by AppSorcery**
these changes allow for a global rating (totalRate) to be displayed in the absence of a personal rating (initialRate) as well as another "bucket" (total) to represent the global rating symbols.

Example (using red/yellow stars for total vs personal rating)
you visit a movie page for a movie you've never rated.  you see a rating in red stars representing what the average user rating is.  you hover over the rating widget and it tracks your movement with yellow stars.  if you click on the yellow stars you save your rating and from that point forward only your rating in yellow stars is ever displayed...

### CODE REVIEW

**Comment by dreyescat** [comment]
Hey @AppSorcery, thanks for contributing! I think I get the idea but I would like to be sure. Could you provide an example of what you mean? The one that you propose would be great. How would you use the `props` you added to implement this use case?

PD:
I have a gut feeling that this use case could be implemented without this third state you propose :wink:. But I want to be sure I understand what you are trying to solve first.

**Comment by AppSorcery** [comment]
I made some changes to the index.html file to utilize the changes.  I think it could be done in the parent component (without any of my changes) if you added an onChange listener which changed the parent state (forcing a re-render) and then you swapped the full image set depending on whether the change is undefined or has a value...

**Comment by dreyescat** [comment]
Thanks @AppSorcery for the examples!

At first I was not really convinced it was a good idea to add this *third state* to the mix. I want to keep this component as simple as possible. Do one thing but do it well. I don't want a component with lots of props.

Then I decided to give it 5 minutes (well, a couple of days actually) and see if I could change my mind. So, well, now I see it different. If we look at it as if it was a sort of *placeholder* rate it makes more sense to me. A placeholder in the same way you have the `input` text [placeholder](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-placeholder) attribute. What you describe looks like the same behaviour but applied to a rating. The placeholder rate is shown only if no value is selected and no mouse hover.

File: src/Rating.jsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ var Rating = React.createClass({
     stop: React.PropTypes.number,
     step: React.PropTypes.number,
     initialRate: React.PropTypes.number,
+    totalRate: React.PropTypes.number,
```
Comment (dreyescat) [inline src/Rating.jsx]: What do you think if we change it to `placeholderRate`?

File: src/Rating.jsx [hunk 2]
```diff
@@ -40,6 +41,16 @@ var Rating = React.createClass({
       React.PropTypes.string,
       // Style objects.
       React.PropTypes.object]),
+    total: React.PropTypes.oneOfType([
```
Comment (dreyescat) [inline src/Rating.jsx]: And this to `placeholder`?

File: src/Rating.jsx [hunk 3]
```diff
@@ -141,10 +170,15 @@ var Rating = React.createClass({
       // indexes NaN or above the last full index.
       var percent = i - lastFullIndex === 0 ? index % 1 * 100 :
         i - lastFullIndex < 0 ? 100 : 0;
+
+	  var icon = (!this.state.selected &&
```
Comment (dreyescat) [inline src/Rating.jsx]: Shouldn't `var icon` be outside the `for..loop`? I don't see any reason to calculate it on each iteration.

**Comment by dreyescat** [comment]
Ok. I have made a quick code review and put some suggestions.

If it is not much to ask for, could you:
  1. Remove changes in `lib` folder. You do not need to commit changes under `lib`. They are autom    atically created when the version is released. You only need to commit changes under `src` folder.
  1. Squash all commits into a single one.
  1. And if you really feel generous... could you update the documentation with these two new props? It is in the [README.md](https://github.com/dreyescat/react-rating/blob/master/README.md#properties).

**Comment by AppSorcery** [comment]
- I've renamed the "total" stuff to "placeholder" stuff as requested (i do like "placeholder" better btw)
- removed lib changes
- squashed commits
- updated the doc; however you might want to review the wording I'm not entirely happy with it.

**Comment by dreyescat** [comment]
Wow! That's fast!
The wording is Ok for me.

Thanks!

**Pull request merged**